### PR TITLE
Test Infrastructure updates

### PR DESCRIPTION
**Pull request by MaceWindu**

### CODE REVIEW

File: Tests/Linq/Linq/DateTimeFunctionsTests.cs [hunk 1]
```diff
@@ -278,13 +278,14 @@ public void DatePartWeekNumberingType([DataSources(false)] string context)
...
-				var muslimWeeks           = new[] { 52, 53, 53, 53, 1, 1, 1, 1, 2, 2, 2, 2 };
-				var primitive             = new[] { 52, 52, 52, 53, 1, 1, 1, 1, 1, 1, 1, 2 };
+				var isoWeeks                     = new[] { 52, 52, 52, 53, 1, 1, 1, 1, 1, 1, 2, 2 };
+				var sqliteParodyNumbering        = new[] { 52, 52, 52, 53, 0, 0, 0, 0, 0, 0, 1, 1 };
+				var isoProperWeeks               = new[] { 52, 52, 52,  1, 1, 1, 1, 1, 1, 1, 2, 2 };
+				var usWeeks                      = new[] { 52, 52, 53, 53, 1, 1, 1, 1, 1, 2, 2, 2 };
+				var usWeeksZeroBased             = new[] { 51, 51, 52, 52, 0, 0, 0, 0, 0, 1, 1, 1 };
+				var muslimWeeks                  = new[] { 52, 53, 53, 53, 1, 1, 1, 1, 2, 2, 2, 2 };
+				var primitive                    = new[] { 52, 52, 52, 53, 1, 1, 1, 1, 1, 1, 1, 2 };
+				var usWeeksZeroBased_byDayOfYear = new[] { 51, 51, 52, 52, 0, 0, 0, 0, 0, 0, 1, 1 };
```
Comment (MaceWindu) [inline Tests/Linq/Linq/DateTimeFunctionsTests.cs:288]: @nsarris , I've checked it. Are you sure it correct? 6 Jan should be week 1, not week 0 in US numbering.

BTW, I don't see US numbering support here https://www.ibm.com/support/knowledgecenter/ssw_ibm_i_74/db2/rbafzscavarcharformat.htm, and week numering starts from 1 
I see IW (or WEEK_ISO function) for ISO numbering and WW (or WEEK function) for numbering called `primitive` in test

I've checked your implementation https://github.com/LinqToDB4iSeries/Linq2DB4iSeries/blob/30ec3f748cb0246f7f39e94336fcb1fd0209dcd7/Source/ISeriesProvider/DB2iSeriesMetadataReader.cs#L169, why don't you use WEEK/WEEK_ISO functions?

Comment (nsarris) [reply Tests/Linq/Linq/DateTimeFunctionsTests.cs:288]: Didn't put much thought behind it, I was focused an getting all tests to run, perhaps I mssed something. I'll check it again this weekend.

Comment (nsarris) [reply Tests/Linq/Linq/DateTimeFunctionsTests.cs:288]: Ok I checked it and you are right, I don't know why Roy (the original author of the provider) opted for Day of Year / 7, but that's what I found was going on and I didn't think to check the function declaration. I tested both Week and Week_ISO and both worked as expected. I'm keeping Week. You can remove that change.

**Pull request merged**